packages/module/src/WarningModal/WarningModal.tsx

```diff
@@ -69,17 +92,36 @@ const WarningModal: React.FunctionComponent<WarningModalProps> = ({
       ouiaId={ouiaId}
       {...props}
     >
-      {children}
-      {withCheckbox ? (
-        <Checkbox
-          isChecked={checked}
-          onChange={(_event, value) => setChecked(value)}
-          label={checkboxLabel}
-          id="warning-modal-check"
-          className="pf-v6-u-mt-lg"
-          ouiaId={`${ouiaId}-confirm-checkbox`}
-        />
-      ) : null}
+      <Stack hasGutter>
+        <StackItem>{children}</StackItem>
+        <StackItem>
+          {confirmationInputProps ? (
+            <Flex direction={{ default: 'column' }} spaceItems={{ default: 'spaceItemsSm' }}>
+              <FlexItem>
+                {confirmationInputProps && confirmationInputLabel(deleteNameSanitized)}
+              </FlexItem>
+              <TextInput
+                ouiaId={`${ouiaId}-confirmation-text-input`}
+                value={inputValue}
+                onChange={(_e, value) => setInputValue(value)}
+                {...confirmationInputProps}
```

Comment:
@InsaneZein for the default props mentioned above, we could do this. 
https://github.com/patternfly/react-component-groups/pull/540#discussion_r1909146969
What do you think?
```suggestion
                {...{ type: 'text', isRequired: true, ...confirmationInputProps}}
```